source/FreeRTOS_DNS.c

```diff
@@ -419,14 +419,14 @@
     static uint32_t prvDNSReply( struct xDNSBuffer * pxReceiveBuffer,
                                  TickType_t uxIdentifier )
     {
-        uint32_t ulIPAddress = 0UL;
+        uint32_t ulIPAddress = ( uint32_t ) 0UL;
         BaseType_t xExpected;
         const DNSMessage_t * pxDNSMessageHeader =
             ( ( const DNSMessage_t * )
               pxReceiveBuffer->pucPayloadBuffer );
 
         /* See if the identifiers match. */
-        xExpected = ( uxIdentifier == ( TickType_t ) pxDNSMessageHeader->usIdentifier );
+        xExpected = ( BaseType_t ) ( uxIdentifier == ( TickType_t ) pxDNSMessageHeader->usIdentifier );
```

Comment:
My coverity doesn't like an expression of the type:
~~~c
    int same = ( a == b );
~~~
If can either be solved in two ways:
~~~c
    int same;
    if( a == b )
    {
        same = pdTRUE;
    }
    else
    {
        same = pdFALSE;
    }
~~~
And this one-liner also works :
~~~
    int same = ( a == b ) ? pdTRUE : pdFALSE;
~~~